dandiapi/api/doi.py

```diff
@@ -24,22 +24,81 @@ def doi_configured() -> bool:
     return any(setting is not None for setting, _ in DANDI_DOI_SETTINGS)
 
 
-def _generate_doi_data(version: Version):
+# TODO(asmacdo) findable dois
+# def _generate_doi_data(version: Version, event: str):
+def _generate_doi_data(version: Version, version_doi=True):
     from dandischema.datacite import to_datacite
 
     publish = settings.DANDI_DOI_PUBLISH
     # Use the DANDI test datacite instance as a placeholder if PREFIX isn't set
     prefix = settings.DANDI_DOI_API_PREFIX or '10.80507'
     dandiset_id = version.dandiset.identifier
     version_id = version.version
-    doi = f'{prefix}/dandi.{dandiset_id}/{version_id}'
+    if version_doi:
+        doi = f'{prefix}/dandi.{dandiset_id}/{version_id}'
+    else:
+        doi = f'{prefix}/dandi.{dandiset_id}'
     metadata = version.metadata
     metadata['doi'] = doi
-    return (doi, to_datacite(metadata, publish=publish))
+    # TODO(asmacdo) findable dois
+    # datacite_body = to_datacite(metadata, event=event)
+    datacite_body = to_datacite(metadata, version_doi=version_doi)
+    return (doi, datacite_body)
+
+def update_dandiset_doi(version: Version) -> str:
```

Comment:
Maybe this whole chunk is unnecessary?  https://github.com/dandi/dandi-archive/issues/1709#issuecomment-1924227635

If we can just point to the Dandiset DOI to the DLP, rather than the latest version then we dont need to update.